getBlocksForRange.js

```diff
@@ -33,11 +33,36 @@ function intToHex(int) {
   return '0x' + hexString
 }
 
-function query(provider, method, params) {
+function sendAsync(provider, request) {
   return new Promise((resolve, reject) => {
-    provider.sendAsync({ id: 1, jsonrpc: '2.0', method, params }, (err, res) => {
-      if (err) return reject(err)
-      resolve(res.result)
-    })
-  })
+    provider.sendAsync(request, (error, response) => {
+      if (error) {
+        reject(error);
+      } else if (response.error) {
+        reject(response.error);
+      } else if (response.result) {
+        resolve(response.result);
+      } else {
+        reject(new Error("Result was empty"));
+      }
+    });
+  });
+}
+
+async function query(provider, method, params) {
+  for (let i = 0; i < 3; i++) {
+    try {
+      return await sendAsync(provider, {
+        id: 1,
+        jsonrpc: "2.0",
+        method,
+        params,
+      });
+    } catch (error) {
+      console.error(
+        `provider.sendAsync failed: ${error.stack || error.message || error}`
+      );
+    }
+  }
+  throw new Error(`Block not found for params: ${params}`);
```

Comment:
[nit] The params in the error message seem like they get oversimplified. What about `JSON.stringify(params)`?

Reply:
Good point @mcmire